### PR TITLE
[New Model] VenusREM

### PR DESCRIPTION
**Pull request by tyang816**
> I'm very sorry, but I have changed the model's name again according to my supervisor's requirements.

Hi, Pascal

We recently fused MSA into the ProSST model and achieved better performance, achieving a score of **0.518** on substitutions named [VenusREM](https://github.com/tyang816/VenusREM). (https://arxiv.org/abs/2410.21127)

1. Added the scoring code of *venusrem* in `proteingym/baselines/venusrem`.
2. Added *venusrem* information to `config.json` and `constants.json`.
3. Added  `scripts/scoring_DMS_zero_shot/scoring_VenusREM_substitutions.sh`.
4. Update the structure tokenizer of *ProSST*.

It is worth noting:
1. A new conda environment is required, the configuration file is in `proteingym/baselines/venusrem/environment.yaml` (same as **ProtSSN** and **ProSST**), or you can use the environment of **ProSST**.
2. The original PDB files are same with **ProtSSN** and **ProSST**, you can [download](https://lianglab.sjtu.edu.cn/files/ProtSSN-2024/proteingym_v1_pdb.tar.gz) them here. The .a2m files can be [downloaded](https://huggingface.co/datasets/tyang816/VenusREM/resolve/main/aa_seq_aln_a2m.tar.gz) here.
3. More model information can be seen in [VenusREM repository](https://github.com/tyang816/VenusREM).

Thanks again for your community efforts.

best wishes
Yang Tan

### CODE REVIEW

**Comment by pascalnotin** [comment]
Hi @tyang816 -- thank you for the PR, and congrats on the new work!

I finally had the chance to read the VenusREM paper, and should be able to merge into PG relatively soon. I had a few questions / comments in the meantime:
1. Is the model combining sequence & structure with which you obtain the predicted native logits (ie. O^{native}_{iv} in Eq.6) different from ProSST?
2. I believe the first paper to combine sequence, structure and alignment was actually this work :) --> https://openreview.net/forum?id=8PbTU4exnV 
3. Regarding the homology retrieval component, is it different from the retrieval at inference introduced in Tranception (https://proceedings.mlr.press/v162/notin22a.html)?  

Kind regards,
Pascal

**Comment by pascalnotin** [comment]
One additional question: are the files made available at https://huggingface.co/datasets/tyang816/VenusREM/blob/main/aa_seq_aln_a2m.tar.gz identical to the MSAs from the ProteinGym repo, or did you apply additional filtering / transformations on them?

**Comment by tyang816** [comment]
Hi Pascal,

Thank you for merging our work into the main branch! We’re excited to see the update reflected on the leaderboard soon.

Regarding your questions:

1. The model combining sequence and structure for predicting native logits (i.e., 𝑂𝑖𝑣 in Eq. 6) is identical to ProSST—there are no differences.
2. We appreciate your recommendation and apologize for missing this relevant work. We'll read the paper carefully.
3. Our approach is very similar to Tranception. The main difference is that Tranception applies constraints when generating amino acid tokens, while we directly combine the two probabilities. Additionally, we explicitly discuss the impact of EVcouplings, Foldseek, and ColabFold on zero-shot scoring for homologous sequences.
4. The original files are unchanged except for renaming some based on the DMS name (some files are duplicates). Any additional transformations, such as handling gaps or case sensitivity, are taken care of in the code.

Let us know if you have any further questions!

Best, 
Yang Tan

**Comment by pascalnotin** [comment]
Hi @tyang816 - thank you for the clarifications! I updated the performance files on the repo yesterday (see for example [here](https://github.com/OATML-Markslab/ProteinGym/blob/main/benchmarks/DMS_zero_shot/substitutions/Spearman/Summary_performance_DMS_substitutions_Spearman.csv)), and we will update the website in the coming week.
Congrats again!

**Comment by tyang816** [comment]
Thanks! We'll continue following your work—it has been a great source of inspiration for us.😊
